nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -0,0 +1,230 @@
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Test
+    public void shouldMoveBlobDocumentToColdStorage() throws IOException {
+        DocumentModel documentModel = createDocument(DEFAULT_DOC_NAME, true);
+
+        // move the blob to cold storage
+        documentModel = ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+        transactionalFeature.nextTransaction();
+        documentModel = session.getDocument(documentModel.getRef());
+
+        assertTrue(documentModel.hasFacet(FacetNames.COLD_STORAGE));
+
+        assertNull(documentModel.getPropertyValue(ColdStorageHelper.FILE_CONTENT_PROPERTY));
+
+        // check if the `coldstorage:coldContent` contains the original file content
+        checkBlobContent(documentModel, ColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY, FILE_CONTENT);
+    }
+
+    @Test
+    public void shouldFailWhenMovingDocumentBlobAlreadyInColdStorage() {
+        DocumentModel documentModel = createDocument(DEFAULT_DOC_NAME, true);
+
+        // move for the first time
+        documentModel = ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+
+        // try to make another move
+        try {
+            ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+            fail("Should fail because the content is already in cold storage");
+        } catch (NuxeoException ne) {
+            assertEquals(SC_CONFLICT, ne.getStatusCode());
+            assertEquals(String.format("The main content for document: %s is already in cold storage.", documentModel),
+                    ne.getMessage());
+        }
+    }
+
+    @Test
+    public void shouldFailWhenMovingToColdStorageDocumentWithoutContent() {
+        DocumentModel documentModel = createDocument(DEFAULT_DOC_NAME, false);
+        try {
+            ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+            fail("Should fail because there is no main content associated with the document");
+        } catch (NuxeoException ne) {
+            assertEquals(SC_NOT_FOUND, ne.getStatusCode());
+            assertEquals(String.format("There is no main content for document: %s.", documentModel), ne.getMessage());
+        }
+    }
+
+    @Test
+    public void shouldRetrieveDocumentBlobFromColdStorage() {
+        DocumentModel documentModel = createDocument(DEFAULT_DOC_NAME, true);
+
+        // move the blob to cold storage
+        ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+
+        // retrieve, which means initiate a request to restore the blob from cold storage
+        documentModel = ColdStorageHelper.retrieveContentFromColdStorage(session, documentModel.getRef(),
+                NUMBER_OF_DAYS_OF_AVAILABILITY);
+        transactionalFeature.nextTransaction();
+        documentModel = session.getDocument(documentModel.getRef());
```

Comment:
So, finally, don't we want to use `refresh` instead of `session.getDocument`?

Reply:
Let me try again, because initially (after the first review) i tried it and i had exception.